### ISSUE TITLE
Bug: no-unused-expression false positive

### ISSUE THREAD

**Issue by rbalicki2** [eslint/eslint#19428]
### Environment

-

### What parser are you using?

Default (Espree)

### What did you do?

-

### What did you expect to happen?

```
foo`asdf`;
```
is incorrectly categorized as an unused expresion for the purposes of `no-unused-expressions`. This is a function call, and should be treated identically to
```
foo();
```

### What actually happened?

-

### Link to Minimal Reproducible Example

-

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by yeonjuan** [comment]
Hi @rbalicki2  Setting the `allowTaggedTemplates` option to `false` is probably what you're looking for.

- https://eslint.org/docs/latest/rules/no-unused-expressions#allowtaggedtemplates

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLWV4cHJlc3Npb25zOiBbXCJlcnJvclwiLCB7IFwiYWxsb3dUYWdnZWRUZW1wbGF0ZXNcIjogZmFsc2UgfV0qL1xuXG50YWdgc29tZSB0YWdnZWQgdGVtcGxhdGUgc3RyaW5nYDsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=)
```js
/*eslint no-unused-expressions: ["error", { "allowTaggedTemplates": false }]*/

tag`some tagged template string`; // Error
```

**Comment by mdjermanovic** [comment]
> ```
> foo`asdf`;
> ```
> 
> 
>     
>   
> 
> is incorrectly categorized as an unused expresion for the purposes of `no-unused-expressions`. This is a function call

This is indeed technically a function call but typically doesn't have side effects, so the rule by default reports it. You can change this behavior by setting the `allowTaggedTemplates` option to `true`.

Closing as this works as intended.
